src/main/java/it/gov/pagopa/pu/send/service/SendServiceImpl.java

```
@@ -0,0 +1,52 @@
+package it.gov.pagopa.pu.send.service;
+
+import it.gov.pagopa.pu.send.connector.client.SendClientImpl;
+import it.gov.pagopa.pu.send.connector.send.generated.dto.PreLoadRequestDTO;
+import it.gov.pagopa.pu.send.connector.send.generated.dto.PreLoadResponseDTO;
+import it.gov.pagopa.pu.send.enums.FileStatus;
+import it.gov.pagopa.pu.send.enums.NotificationStatus;
+import it.gov.pagopa.pu.send.model.SendNotification;
+import it.gov.pagopa.pu.send.repository.SendNotificationRepository;
+import it.gov.pagopa.pu.send.util.NotificationUtils;
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+public class SendServiceImpl implements SendService{
+
+  private final SendNotificationRepository sendNotificationRepository;
+  private final SendClientImpl sendClient;
+
+  public SendServiceImpl(SendNotificationRepository sendNotificationRepository,
+    SendClientImpl sendClient) {
+    this.sendNotificationRepository = sendNotificationRepository;
+    this.sendClient = sendClient;
+  }
+
+  @Override
+  public void preloadFiles(String sendNotificationId) {
+    SendNotification notification = sendNotificationRepository.findById(sendNotificationId)
+      .orElseThrow(() -> new IllegalArgumentException("Notification not found with id: " + sendNotificationId));
```

if the exception is not handled, it will return to the client as a 500

---

handled with a 404 http status as declared into openapi